src/Ocelot/DownstreamRouteFinder/UrlMatcher/UrlPathPlaceholderNameAndValueFinder.cs

```diff
@@ -104,6 +109,11 @@ private static bool NothingAfterFirstForwardSlash(string path)
 
         private static string GetPlaceholderValue(string urlPathTemplate, string query, string variableName, string urlPath, int counterForUrl, char delimiter)
         {
+            if(counterForUrl > urlPath.Length)
+            {
+                return "";
+            }
```

Comment:
I that a border case?
Which one?
If counter reached the end of url then no plaveholder values?

Reply:
Yes exactly this is the way to circuit break any empty value in the end of the path for all the cases